legal-api/src/legal_api/models/legal_entity.py

```diff
@@ -479,11 +479,43 @@ def business_name(self):
         if not self.is_firm:
             return self._legal_name
 
-        if alternate_name := self.alternate_names.first():
+        if alternate_name := self._alternate_names.filter_by(identifier=self.identifier).one_or_none():
             return alternate_name.name
 
         return None
 
+    @property
+    def alternate_names(self):
+        """Return operating names for a business if any."""
+        le_alias = aliased(LegalEntity)
+        alternate_names = (
+            db.session.query(AlternateName.identifier,
+                             AlternateName.name,
+                             AlternateName.start_date,
+                             le_alias.entity_type,
+                             le_alias.founding_date)
```

Comment:
Is end_date not included, as we're deleting the AlternateName when it is ended?

Reply:
This is how I've converted the existing data right now.  

![image](https://github.com/bcgov/lear/assets/6685223/e1cadea9-1711-463c-a534-88fe8ee156f6)

I think maybe each of these history rows should have a unique id and two corresponding history entries.  One with the initial entry and the end-dated entry.  And the end-dated entries would not be in `alternate_names`.  If we want this, I'll need to make some updates to the conversion scripts.

Both approaches would mean that `alternate_names` will only contain alternate name entries that were not end-dated.

Reply:
Cool, pretty sure that's what we agreed to, and is consistent with the rest of the history table.
I like it!